lib/game/base_game.dart

```diff
@@ -67,9 +67,13 @@ class BaseGame extends Game with FPSCounter {
   }
 
   /// Prepares and registers a component to be added on the next game tick
-  void add(Component c) {
+  ///
+  /// This methods is an async operation since it await the `onLoad` method of the component. Nevertheless, this method only need to be waited to finish if by some reason, your logic needs to be sure that the component has finished loading, otherwise, this method can be called without waiting for it to finish as the BaseGame already handle the loading of the component.
```

Comment:
so I wonder if this can cause issues if I don't want it to be async. even if I return a completed future, it would not add immediately. I don't know exactly how dart handles threads but maybe we provide an option to bypass this?

Reply:
We could do something that might be considered slightly hacky, and that is having a static future function defined on the component which is just empty and have that one as the default implementation and we have a small method on the component that compares the current implementation of `onLoad` and that static default function which basically returns whether `onLoad` has been overridden or not.

```dart
abstract class Component {
    static _defaultOnLoad = () => Future.value();

    Future<void> onLoad() = _defaultOnLoad;

    bool get needsLoading => onLoad != _defaultOnLoad; 
}
```

Reply:
I have implemented a new idea, I think it work nice, I also added some test to ensure that things are being added right away when it is possible.

Reply:
Aaah, much better than my solution!